src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/RedisTransportFactoryTest.php

```diff
@@ -28,22 +28,59 @@ public function testSupportsOnlyRedisTransports()
 
         $this->assertTrue($factory->supports('redis://localhost', []));
         $this->assertTrue($factory->supports('rediss://localhost', []));
+        $this->assertTrue($factory->supports('redis:?host[host1:5000]&host[host2:5000]&host[host3:5000]&sentinel_master=test&dbindex=0', []));
         $this->assertFalse($factory->supports('sqs://localhost', []));
         $this->assertFalse($factory->supports('invalid-dsn', []));
     }
 
     /**
      * @group integration
+     *
+     * @dataProvider createProvider
      */
-    public function testCreateTransport()
+    public function testCreateTransport(string $dsn, array $options = [])
     {
         $this->skipIfRedisUnavailable();
 
         $factory = new RedisTransportFactory();
         $serializer = $this->createMock(SerializerInterface::class);
-        $expectedTransport = new RedisTransport(Connection::fromDsn('redis://'.getenv('REDIS_HOST'), ['stream' => 'bar', 'delete_after_ack' => true]), $serializer);
 
-        $this->assertEquals($expectedTransport, $factory->createTransport('redis://'.getenv('REDIS_HOST'), ['stream' => 'bar', 'delete_after_ack' => true], $serializer));
+        $this->assertEquals(
+            new RedisTransport(Connection::fromDsn($dsn, $options), $serializer),
+            $factory->createTransport($dsn, $options, $serializer)
+        );
+    }
+
+    /**
+     * @return iterable<array{0: string, 1: array}>
+     */
+    public static function createProvider(): iterable
+    {
+        yield 'scheme "redis" without options' => [
+            'redis://'.getenv('REDIS_HOST'),
+            [],
+        ];
+
+        yield 'scheme "rediss" without options' => [
+            'rediss://'.getenv('REDIS_HOST'),
+            [],
+        ];
+
+        yield 'scheme "redis" with options' => [
+            'redis://'.getenv('REDIS_HOST'),
+            ['stream' => 'bar', 'delete_after_ack' => true],
+        ];
+
+        yield 'scheme "rediss" with options' => [
+            'rediss://'.getenv('REDIS_HOST'),
+            ['stream' => 'bar', 'delete_after_ack' => true],
+        ];
+
+        yield 'redis_sentinel' => [
+            'redis:?host[host1:5000]&host[host2:5000]&host[host3:5000]&sentinel_master=test&dbindex=0',
```

Comment:
as we used `getenv('REDIS_HOST')` before I think we need a real connection and I am not sure if we can connect using this sentinel stuff 🤔

Reply:
Sentinel has its own protocol, which is sort of a telnet connection (as well as the redis core protocol). But here we just connect to ask for the current master in the cluster which, since sentinel can handle multiple clusters, is set in the `sentinel_master` parameter. Sentinel returns an IP and we connect to that IP with the usual Redis protocol.
So if you want to connect for a `PING`, you can do so to a sentinel as well. Almost all other commands differ and would require to setup a sentinel instance.

Reply:
Can you please adjust my code to make it work? I have never worked with sentinel

Reply:
The sentinel code is fixed, but the `rediss` schemes still fail. I barely found any ssl tests. Any idea @OskarStark? We may have to add another process (with ssl enabled) to the pipeline...?

Reply:
@alexandre-daubois do you know anything related?

Reply:
We don't have Redis Secure (accepting the `rediss` scheme) setup in the integration workflow. I'll try to have a look at it, it would be interesting to add it! But that would definitely be for a separate PR. I think we can pass on this case :man_shrugging:

Reply:
We don't have Redis Secure (accepting the `rediss` scheme) setup in the integration workflow. I'll try to have a look at it, it would be interesting to add it! But that would definitely be for a separate PR. I think we can pass on this case :man_shrugging:

Reply:
ok lets remove it then

Reply:
Removed.